crengine/src/private/lvfreetypefontman.cpp

```diff
@@ -907,14 +976,12 @@ bool LVFreeTypeFontManager::RegisterDocumentFont(int documentId, LVContainerRef
         if (familyName == "Times" || familyName == "Times New Roman")
             fontFamily = css_ff_serif;
 
-        bool boldFlag = !faceName.empty() ? bold : (face->style_flags & FT_STYLE_FLAG_BOLD) != 0;
-        bool italicFlag = !faceName.empty() ? italic : (face->style_flags & FT_STYLE_FLAG_ITALIC) !=
-                                                       0;
+        bool italicFlag = !faceName.empty() ? italic : (face->style_flags & FT_STYLE_FLAG_ITALIC) != 0;
 
         LVFontDef def(
                 name8,
                 -1, // height==-1 for scalable fonts
-                boldFlag ? 700 : 400,
+                getFontWeight(face),
```

Comment:
Shouldn't you do the same in the similar bit of code in LVFreeTypeFontManager::SetAlias() line 589 ?

Reply:
Similar bit of code in LVFreeTypeFontManager::RegisterExternalFont() - but I guess in there, you trust the "bold" parameter, although you also get a face name.

Reply:
I just didn't know what it was. As far as I understand this is for embedded epub fonts. 
Something like this?
```cpp
diff --git a/crengine/src/private/lvfreetypefontman.cpp b/crengine/src/private/lvfreetypefontman.cpp
index db16dd881..f93b79774 100644
--- a/crengine/src/private/lvfreetypefontman.cpp
+++ b/crengine/src/private/lvfreetypefontman.cpp
@@ -652,13 +652,13 @@ bool LVFreeTypeFontManager::SetAlias(lString8 alias, lString8 facename, int id,
             fontFamily = css_ff_serif;
         */
 
-        bool boldFlag = !facename.empty() ? bold : (face->style_flags & FT_STYLE_FLAG_BOLD) != 0;
+        int weight = !facename.empty() ? (bold ? 700 : 400) : getFontWeight(face);
         bool italicFlag = !facename.empty() ? italic : (face->style_flags & FT_STYLE_FLAG_ITALIC) != 0;
 
         LVFontDef def2(
                 item->getDef()->getName(),
                 -1, // height==-1 for scalable fonts
-                boldFlag ? 700 : 400,
+                weight,
                 italicFlag,
                 -1, // OpenType features = -1 for not yet instantiated fonts
                 fontFamily,
```
Or ignore the "bold" argument completely? He seems to come from here `EmbeddedFontStyleParser` (`epubfmt.cpp`).

Reply:
> As far as I understand this is for embedded epub fonts.

Yes, I think so too, but I'm really not familiar with all that.

> Something like this?

Yes, I guess so, this is the most conservative/similar to the previous behaviour.

> Or ignore the "bold" argument completely? 

I don't think we should. These must be defined by some CSS @font-face where the publisher decides what font to use for regular and bold. I don't think we should trust that much the name it gives there to the font names (although I dunno if we get the font name from the CSS, or from the font face data itself).

Reply:
> Similar bit of code in LVFreeTypeFontManager::RegisterExternalFont() - but I guess in there, you trust the "bold" parameter

You mean _must_ trust the "bold" parameter?

Reply:
Should the final patch be like this?
```diff
diff --git a/crengine/src/private/lvfreetypefontman.cpp b/crengine/src/private/lvfreetypefontman.cpp
index db16dd881..81be6b2b9 100644
--- a/crengine/src/private/lvfreetypefontman.cpp
+++ b/crengine/src/private/lvfreetypefontman.cpp
@@ -652,13 +652,13 @@ bool LVFreeTypeFontManager::SetAlias(lString8 alias, lString8 facename, int id,
             fontFamily = css_ff_serif;
         */
 
-        bool boldFlag = !facename.empty() ? bold : (face->style_flags & FT_STYLE_FLAG_BOLD) != 0;
+        int weight = !facename.empty() ? (bold ? 700 : 400) : getFontWeight(face);
         bool italicFlag = !facename.empty() ? italic : (face->style_flags & FT_STYLE_FLAG_ITALIC) != 0;
 
         LVFontDef def2(
                 item->getDef()->getName(),
                 -1, // height==-1 for scalable fonts
-                boldFlag ? 700 : 400,
+                weight,
                 italicFlag,
                 -1, // OpenType features = -1 for not yet instantiated fonts
                 fontFamily,
@@ -992,12 +992,13 @@ bool LVFreeTypeFontManager::RegisterDocumentFont(int documentId, LVContainerRef
         if (familyName == "Times" || familyName == "Times New Roman")
             fontFamily = css_ff_serif;
 
+        int weight = !facename.empty() ? (bold ? 700 : 400) : getFontWeight(face);
         bool italicFlag = !faceName.empty() ? italic : (face->style_flags & FT_STYLE_FLAG_ITALIC) != 0;
 
         LVFontDef def(
                 name8,
                 -1, // height==-1 for scalable fonts
-                getFontWeight(face),
+                weight,
                 italicFlag,
                 -1, // OpenType features = -1 for not yet instantiated fonts
                 fontFamily,
```

Reply:
I think we should trust `font-weight:bold;`.
If the publisher sets:
```css
@font-face {
 font-family:DejaVuSans;
 font-weight:normal;
 font-style:normal;
 src:url(../Fonts/DejaVuSans-Bold.ttf);
}
@font-face {
 font-family:DejaVuSans;
 font-weight:bold;
 font-style:normal;
 src:url(../Fonts/DejaVuSans-ExtraBlack.ttf);
}
```
he wants the bold to be used for regular/400.
By doing the other option, we would synthetized a thin DejaVuSans-Bold for regular - and use DejaVuSans-Bold for bold, and never DejaVuSans-ExtraBlack - wouldn't we?

Reply:
I'm not sure if I got it right. Does patch https://github.com/buggins/coolreader/pull/274#discussion_r603179167 solve this problem?

Reply:
Well, it looks sane - although I dunno what's the thing with doing it only when !faceName.empty() - and what an empty faceName means. It's all in ldomDocument::registerEmbeddedFonts(), but no time to try understanding it :/

Reply:
Ok, added a commit.